twyla/kubedeploy/kube.py

```diff
@@ -20,56 +25,80 @@ def __init__(self,
         self.printer = printer
         self.error_printer = error_printer
         self.deployment_name = deployment_name
+        self.deployment_template = deployment_template or 'deployment.yml'
+
+
+    def parse_deployment_data(self, data):
+        api_client = kubernetes.client.ApiClient()
+
+        # Be like Response, my friend!
+        # Implements res.data to make the deserializer work.
+        class Res:
+            def __init__(self, data):
+                self.data = data
+
+        res = Res(data=data)
+
+        return api_client.deserialize(res, 'ExtensionsV1beta1Deployment')
+
+    def load_deployment_from_file(self):
+        with open(self.deployment_template) as fd:
+            dict_data = yaml.load(fd)
+            return self.parse_deployment_data(json.dumps(dict_data))
```

Comment:
A minor point: I usually indent the return statement out of the with context to signal that it doesn't belong in there, but is the highest level return of the function.

Reply:
Will change.